### PR TITLE
Remove the GTS BUFR2IODA part of the snow obs prep job

### PR DESCRIPTION
**Pull request by jiaruidong2017**
This PR removed the GTS BUFR2IODA part of the snow obs prep job, and replaced it with a direct read from BUFR snow data at runtime by the JEDI executable. 

This PR contributes to https://github.com/NOAA-EMC/global-workflow/pull/2900

### CODE REVIEW

File: parm/snow/obs/config/bufr2ioda_mapping.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bufr2ioda:
+     sfcsno: '{{ HOMEgfs }}/sorc/gdas.cd/test/testinput/bufr_sfcsno_mapping.yaml'
```
Comment (aerorahul) [inline parm/snow/obs/config/bufr2ioda_mapping.yaml.j2]: This file needs to be linked to `PARMgfs/parm/gdas/` instead of being in `sorc/gdas.cd`

Comment (jiaruidong2017) [reply parm/snow/obs/config/bufr2ioda_mapping.yaml.j2]: Done. Thanks @aerorahul

File: parm/snow/obs/config/bufr_sfcsno_mapping.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# (C) Copyright 2021-2022 NOAA/NWS/NCEP/EMC
```
Comment (aerorahul) [inline parm/snow/obs/config/bufr_sfcsno_mapping.yaml]: 2024?

**Comment by CoryMartin-NOAA** [comment]
This should (at least partially) close https://github.com/NOAA-EMC/GDASApp/issues/1279

File: parm/snow/obs/config/sfcsno_snow.yaml.j2 [hunk 3]
```diff
@@ -5,12 +5,13 @@
...
-        type: H5File
-        obsfile: '{{ DATA }}/obs/{{ OPREFIX }}sfcsno_snow.nc4'
+        type: bufr
+        obsfile: '{{ DATA }}/obs/{{ OPREFIX }}sfcsno.tm00.bufr_d'
+        mapping file: '{{ DATA }}/obs/bufr_sfcsno_mapping.yaml'
     obsdataout:
       engine:
         type: H5File
-        obsfile: '{{ DATA }}/diags/diag_sfcsno_snow.nc4'
+        obsfile: '{{ DATA }}/obs/{{ OPREFIX }}sfcsno_snow.nc4'
```
Comment (CoryMartin-NOAA) [inline parm/snow/obs/config/sfcsno_snow.yaml.j2]: this should still be `diags` yes?
```suggestion
        obsfile: '{{ DATA }}/diags/diag_sfcsno_snow.nc4'
```

Comment (jiaruidong2017) [reply parm/snow/obs/config/sfcsno_snow.yaml.j2]: Yes, and I will change back. Thanks.

**Review by CoryMartin-NOAA — COMMENTED** [review]
Do we still need https://github.com/NOAA-EMC/GDASApp/blob/develop/parm/snow/prep/prep_gts.yaml.j2?

**Comment by jiaruidong2017** [comment]
Removed prep_gts.yaml.j2. Thanks @CoryMartin-NOAA

**Review by CoryMartin-NOAA — DISMISSED** [review]

**Comment by CoryMartin-NOAA** [comment]
We should wait to merge until the g-w PR is ready

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**